### PR TITLE
[konflux] skip-checks by default, replaces dots with dashes

### PR DESCRIPTION
**Pull request by ashwindasr**
Changes in this PR:
- Skip checks by default for now (since its failing the pipeline runs)
- Fix issue with successful / failed builds
- Replace dots with dashes, since openshift doesn't support dots

### CODE REVIEW

File: doozer/doozerlib/backend/konflux_image_builder.py [hunk 1]
```diff
@@ -273,6 +274,8 @@ def _new_pipelinerun(generate_name: str, application_name: str, component_name:
         for param in obj["spec"]["params"]:
             if param["name"] == "output-image":
                 param["value"] = output_image
+            if param["name"] == "skip-checks":
+                param["value"] = "true"
```
Comment (ashwindasr) [inline doozer/doozerlib/backend/konflux_image_builder.py:278]: @vfreex this doesn't seem to be working, not sure if you have any idea?

File: doozer/doozerlib/backend/konflux_image_build_pipelinerun.yaml [hunk 2]
```diff
@@ -65,7 +65,7 @@ spec:
       description: Force rebuild image
       name: rebuild
       type: string
-    - default: "false"
+    - default: "true"
```
Comment (ashwindasr) [inline doozer/doozerlib/backend/konflux_image_build_pipelinerun.yaml:68]: doing this worked, but ideally its better to not edit here

File: doozer/doozerlib/backend/rebaser.py [hunk 3]
```diff
@@ -216,6 +216,14 @@ def _resolve_parents(self, metadata: ImageMetadata, dfp: DockerfileParser, image
         for builder in image_from.get("builder", []):
             parents.append(builder)
         parents.append(image_from)
+
+        # Remove duplicates
+        temp = []
+        for parent in parents:
+            if parent not in temp:
+                temp.append(parent)
+        parents = temp
```
Comment (ashwindasr) [inline doozer/doozerlib/backend/rebaser.py:225]: @vfreex Looks like one duplicate was getting into the list, so removed it, but if there is a better way we can refactor to that

Comment (ashwindasr) [reply doozer/doozerlib/backend/rebaser.py:225]: hmm actually this failed rebase for a couple of images (job [run](https://art-jenkins.apps.prod-stable-spoke1-dc-iad2.itup.redhat.com/job/aos-cd-builds/job/build%252Focp4-konflux/42/console))

File: doozer/doozerlib/backend/konflux_image_builder.py [hunk 4]
```diff
@@ -220,6 +222,7 @@ def _new_component(name: str, application: str, component_name: str,
                 "name": name,
                 "annotations": {
                     "build.appstudio.openshift.io/pipeline": '{"name":"docker-build-multi-platform-oci-ta","bundle":"latest"}',
+                    "build.appstudio.openshift.io/status": '{"pac":{"state":"disabled"}}',
```
Comment (ashwindasr) [inline doozer/doozerlib/backend/konflux_image_builder.py:225]: Konflux by default seems to be enabling custom build pipelines by default, causing it to raise PRs (like [this](https://github.com/openshift-priv/oauth-proxy/pull/1)). Was looking at annotations, and this seem to be the difference between one that is enabled and one that is disabled.

File: doozer/doozerlib/backend/konflux_image_builder.py (hunk 1 above)
Comment (vfreex) [reply doozer/doozerlib/backend/konflux_image_builder.py:280]: ship-checks parameter is not defined under `spec.params` (see line 19 of the yaml template). We can add it as in https://github.com/openshift-eng/art-tools/pull/955/files#diff-e45d70130344e3d45bd59cb091bd0a794121c675ce90f17c5b4e4702ded620ffR281

File: doozer/doozerlib/backend/rebaser.py (hunk 3 above)
Comment (vfreex) [reply doozer/doozerlib/backend/rebaser.py:225]: @ashwindasr I checked the build log and looks like builders are added to the `parents` array twice. Removing line 216-217 should solve this.

**Review by vfreex — COMMENTED** [review]
@ashwindasr Thanks for the PR.  After you address the issues I commented, feel free to merge.

File: doozer/doozerlib/backend/konflux_image_builder.py (hunk 1 above)
Comment (ashwindasr) [reply doozer/doozerlib/backend/konflux_image_builder.py:280]: Thanks!

File: doozer/doozerlib/backend/konflux_image_build_pipelinerun.yaml (hunk 2 above)
Comment (ashwindasr) [reply doozer/doozerlib/backend/konflux_image_build_pipelinerun.yaml:68]: Reverted

File: doozer/doozerlib/backend/rebaser.py (hunk 3 above)
Comment (ashwindasr) [reply doozer/doozerlib/backend/rebaser.py:225]: Thanks!

**Comment by ashwindasr** [comment]
[tested](https://art-jenkins.apps.prod-stable-spoke1-dc-iad2.itup.redhat.com/job/aos-cd-builds/job/build%252Focp4-konflux/45/console), works as expected

**Pull request merged**